src/main/kotlin/com/autonomousapps/tasks/FindDeclaredProcsTask.kt

```diff
@@ -90,30 +91,28 @@ abstract class FindDeclaredProcsTask : DefaultTask() {
   @get:Internal
   abstract val inMemoryCacheProvider: Property<InMemoryCache>
 
-  private val kaptClassLoader: ClassLoader? by lazy {
-    val urls = getKaptArtifactFiles()?.toList()?.map { it.toURI().toURL() }?.toTypedArray()
-    urls?.let { URLClassLoader(urls, javaClass.classLoader) }
-  }
-
-  private val annotationProcessorClassLoader: ClassLoader? by lazy {
-    val urls = getAnnotationProcessorArtifactFiles()?.toList()?.map { it.toURI().toURL() }?.toTypedArray()
-    urls?.let { URLClassLoader(urls, javaClass.classLoader) }
-  }
-
   private val inMemoryCache by lazy { inMemoryCacheProvider.get() }
 
   @TaskAction fun action() {
     val outputFile = output.getAndDelete()
     val outputPrettyFile = outputPretty.getAndDelete()
 
+    val kaptClassLoader = newClassLoader("for-kapt", getKaptArtifactFiles())
+    val apClassLoader = newClassLoader("for-annotation-processor", getAnnotationProcessorArtifactFiles())
+
     val kaptProcs = procs(kaptArtifacts, kaptClassLoader)
-    val annotationProcessorProcs = procs(annotationProcessorArtifacts, annotationProcessorClassLoader)
+    val annotationProcessorProcs = procs(annotationProcessorArtifacts, apClassLoader)
     val procs = kaptProcs + annotationProcessorProcs
 
     outputFile.writeText(procs.toJson())
     outputPrettyFile.writeText(procs.toPrettyString())
   }
 
+  private fun newClassLoader(name: String, files: FileCollection?): ClassLoader? {
+    val urls = files?.toList()?.map { it.toURI().toURL() }?.toTypedArray()
+    return urls?.let { FirstClassLoader(name, urls, javaClass.classLoader) }
```

Comment:
Fix is to use this new custom classloader instead of a `URLClassLoader` directly.